lms/djangoapps/certificates/tests/test_webview_views.py

```diff
@@ -1291,6 +1322,33 @@ def test_certificate_custom_language_template_with_locale_language_from_catalogu
         self.assertEqual(response.status_code, 200)
         self.assertContains(response, 'course name: test_right_lang_template')
 
+    @override_settings(FEATURES=FEATURES_WITH_CUSTOM_CERTS_ENABLED)
+    @ddt.data(True, False)
+    @patch('certificates.views.webview.get_course_run_details')
+    @patch('certificates.api.get_course_organization_id')
+    def test_certificate_custom_template_with_hours_of_effort(self, include_effort, mock_get_org_id, mock_get_course_run_details):
+        """
+        Tests custom template properly retrieves and calculates Hours of Effort when the feature is enabled
+        """
+        # mock the response data from Discovery that updates the context for template lookup and rendering
+        mock_get_course_run_details.return_value = self.mock_course_run_details
+        CertificateGenerationCourseSetting.set_include_hours_of_effort_for_course(self.course.id, include_effort)
+        CertificateGenerationCourseSetting.set_language_specific_templates_enabled_for_course(self.course.id, include_effort)
```

Comment:
It would be good to also have a way to set both of these together so that we do not need two update queries every time we want to create a `CertificateGenerationCourseSetting` object.

Reply:
Do you think a generic set() method that takes a course_key and a dictionary of attribute-value pairings would be ideal?

Reply:
If they're just being used for tests, I'd probably just create the object directly in the test and get rid of the setters, or create a helper method to do it in the test itself.